File: 0283-move-zeroes/0283-move-zeroes.c
void moveZeroes(int* nums, int numsSize) {
    int zeroIndex = -1;  // To store the index of the first zero
    int nonZeroIndex = -1;  // To store the index of the next non-zero element after a zero

    for (int i = 0; i < numsSize; i++) {
        if (nums[i] == 0 && zeroIndex == -1) {
        // First zero found, store its index
        zeroIndex = i;
        }else if (nums[i] != 0 && zeroIndex != -1) {
        // Non-zero element found after zero
        nonZeroIndex = i;
        
        // Now swap the elements at zeroIndex and nonZeroIndex
        int temp = nums[zeroIndex];
        nums[zeroIndex] = nums[nonZeroIndex];
        nums[nonZeroIndex] = temp;
        
        // Update zeroIndex to look for the next zero
        zeroIndex++;
    }
}

}

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        pos = 0  # Position to place the next non-zero element
    
    # Move non-zero elements to the front
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[pos] = nums[i]
                pos += 1
    
    # Fill the rest of the list with zeroes
        for i in range(pos, len(nums)):
            nums[i] = 0
        